src/map.c

```diff
@@ -64,14 +64,7 @@ SEXP map_impl(SEXP env, SEXP type_, SEXP progress, SEXP error_call) {
   SEXPTYPE type = Rf_str2type(CHAR(Rf_asChar(type_)));
 
   SEXP x_val = PROTECT(Rf_eval(x, env));
-
   int n = Rf_length(x_val);
-  if (n == 0) {
-    SEXP out = PROTECT(Rf_allocVector(type, 0));
-    copy_names(x_val, out);
-    UNPROTECT(2);
-    return out;
-  }
 
   // Constructs a call like f(x[[i]], ...) - don't want to substitute
   // actual values for f or x, because they may be long, which creates
```

Comment:
Note that line 110 has an outdated comment
https://github.com/tidyverse/purrr/pull/967/files#diff-67280b99316e68e37d45166adb44ae5ae41cf479e477b8582239d82cbc13070eR110

```
// Check all elements are lists and find recycled length
```